tensorflow_probability/python/experimental/mcmc/sample_fold.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Drivers for streaming reductions framework."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# Dependency imports
+import tensorflow.compat.v2 as tf
+from tensorflow_probability.python.experimental import mcmc
+from tensorflow.python.util import nest
+
+
+__all__ = [
+    'sample_fold',
+]
+
+
+def sample_fold(
+    num_steps,
+    current_state,
+    previous_kernel_results=None,
+    kernel=None,
+    reducers=None,
```

Comment:
For nests, I'm a fan of naming the arguments in the singular (i.e.. `reducer`). Not only is that likely the common situation, it also matches what we do for nests in the rest of TFP (e.g. JointDistribution.event_shape rather than JointDistribution.event_shape**s**).